include/range/v3/view/transform.hpp

```diff
@@ -287,13 +288,17 @@ namespace ranges
                 SizedRange<Rng1 const>() && SizedRange<Rng2 const>())
             constexpr size_type_ size() const
             {
-                return std::min<size_type_>(ranges::size(rng1_), ranges::size(rng2_));
+                return ranges::on(ranges::min,
+                    ranges::compose(convert_to<size_type_>{}, ranges::size))(
+                        rng1_, rng2_);
```

Comment:
> Agreed. :-( Also a shame that we have to type std::equal_to<>{} instead of simply std::equal_to{}. I wonder if the goofy work on class-template-parameter-deduction-from-constructor-arguments helps here any.

[P0091](http://wg21.link/p0091) is purely focused on deduction for class template constructors. The wording is minimal, but it looks to me like `std::equal_to{}` will deduce `void`. It won't help with our `ranges::min<foo>` problem.